Change to simple static site

jusephe opened the pull request:
Previous Create React App has been “serialized” to pure HTML, CSS, and
JavaScript. This is mainly to remove unnecessary abstraction layers and
deployment pipeline while fixing immediate issues with site’s
“index-ability” and “share-ability”.

There are also many minor changes which seemed reasonable given the
amount code changes:

* Using Google Analytics script directly instead of gtag.js. We are not
using other Google marketing tools, so was triggering yet another
request.

* Fix Google Analytics configuration so we don’t have to ask for
  permission to process private information. Previously, we used “store”
  option which seems to no longer exists. Now, we “cookieExpires” which
  does the same things – creates pseudonymous cookie for the duration of
  visitor’s session (instead of default two years).

* Make the site work well without JavaScript and improve accessibility.

  + Keep priorities expanded if JavaScript isn’t available.

  + Set language to Czech.

  + Add additional semantic section and grouping elements for main,
    header and aside parts of the page.

  + Linked labels to respective form fields.

  + Set autofill attributes on form field.

  + Remove useless or inappropriate (therefore confusing) ARIA roles.

  + Set title on inline SVGs to provide textual fallback.

* Don’t provide manifest nor service worker since this is pretty much
  document-oriented website.

* Create new style rules to display expand and collapse icons for policy
  priorities. This is therefore no longer handled by JavaScript, but by
  CSS. We still use collapse component from Bootstrap.

* Use Lato latin web variant (“LatoLatinWeb”) which covers complete
  Czech alphabet but it’s smaller than complete Lato web fonts.

* Update to Bootstrap 4.5 and append use custom minified build (with
  jQuery slim build). The custom build reflects our customization:

  + primary: #6e1646
  + font-family base: Lato (see above)
  + headings-color: primary (see above)

* Update to Font Awesome 5.13, append “svg-with-js” minified style and
  use inline SVGs via sprite. However, we’ve created a custom sprint
  with only a handful of handpicked icons:

  + facebook
  + twitter
  + instagram
  + minus
  + plus

  We don’t use Font Awesome JavaScript code.

* Logo of our movement in SVG.

* Displayed logo with no left-padding to align with the left content
  border.

* Update additionally linked logos (favicon, Open Graph, touch icon).

* Set title based on displayed page

* Add canonical link element and Open Graph meta elements

* As mentioned above, we concatenate our small custom CSS with selected
  Lato web font, Bootstrap custom build and Font Awesome SVG style to
  one CSS file.

* Similarly, we concatenate small custom JS (only sets document
  element’s class to differentiate JS-enabled user-agent in CSS) with
  jQuery slim build and Bootstrap build to one JS file.

jusephe committed:
Change to simple static site

Previous Create React App has been “serialized” to pure HTML, CSS, and
JavaScript. This is mainly to remove unnecessary abstraction layers and
deployment pipeline while fixing immediate issues with site’s
“index-ability” and “share-ability”.

There are also many minor changes which seemed reasonable given the
amount code changes:

* Using Google Analytics script directly instead of gtag.js. We are not
using other Google marketing tools, so was triggering yet another
request.

* Fix Google Analytics configuration so we don’t have to ask for
  permission to process private information. Previously, we used “store”
  option which seems to no longer exists. Now, we “cookieExpires” which
  does the same things – creates pseudonymous cookie for the duration of
  visitor’s session (instead of default two years).

* Make the site work well without JavaScript and improve accessibility.

  + Keep priorities expanded if JavaScript isn’t available.

  + Set language to Czech.

  + Add additional semantic section and grouping elements for main,
    header and aside parts of the page.

  + Linked labels to respective form fields.

  + Set autofill attributes on form field.

  + Remove useless or inappropriate (therefore confusing) ARIA roles.

  + Set title on inline SVGs to provide textual fallback.

* Don’t provide manifest nor service worker since this is pretty much
  document-oriented website.

* Create new style rules to display expand and collapse icons for policy
  priorities. This is therefore no longer handled by JavaScript, but by
  CSS. We still use collapse component from Bootstrap.

* Use Lato latin web variant (“LatoLatinWeb”) which covers complete
  Czech alphabet but it’s smaller than complete Lato web fonts.

* Update to Bootstrap 4.5 and append use custom minified build (with
  jQuery slim build). The custom build reflects our customization:

  + primary: #6e1646
  + font-family base: Lato (see above)
  + headings-color: primary (see above)

* Update to Font Awesome 5.13, append “svg-with-js” minified style and
  use inline SVGs via sprite. However, we’ve created a custom sprint
  with only a handful of handpicked icons:

  + facebook
  + twitter
  + instagram
  + minus
  + plus

  We don’t use Font Awesome JavaScript code.

* Logo of our movement in SVG.

* Displayed logo with no left-padding to align with the left content
  border.

* Update additionally linked logos (favicon, Open Graph, touch icon).

* Set title based on displayed page

* Add canonical link element and Open Graph meta elements

* As mentioned above, we concatenate our small custom CSS with selected
  Lato web font, Bootstrap custom build and Font Awesome SVG style to
  one CSS file.

* Similarly, we concatenate small custom JS (only sets document
  element’s class to differentiate JS-enabled user-agent in CSS) with
  jQuery slim build and Bootstrap build to one JS file.

Co-authored-by: Josef Šupka <josef.supka@gmail.com>

f4z4on commented:
Několik poznámek. Mám rád, když se Git používá k lidskému popisu změn a zachycují se tak rozhodnutí, která za těmi změnami jsou. Z Gitu se potom stává skvělý dokumentační nástroj, který rovnou počítá s tím, že se věci v čase mění, takže není potřeba nic aktualizovat. Člověk ale pořád musí popsat co a proč dělá. Osobně si myslím, že je tisíckrát lepší to dělat přímo v Gitu než tady na GitHubu (už jen kvůli přenositelnosti těch informací jinam) a spíš se víc rozepisovat než méně.

Celkem jsem to PR překopal. Ta Create React Appka tahala nějaké věci, které nepotřebovala. Tak jsem se smazal. Když už jsem byl v tom, tak jsem aktualizoval ty knihovny třetích stran (Bootstrap a Font Awesome). Když jsem aktualizoval Font Awesome, tak jsem změnil způsob, jakým ty ikony načítám (používám inline SVG, ať zdroj vizuální informace není v HTML). No, a když už jsem byl v tom 😅 tak jsem si řekl, že se podívám na přístupnost toho webu. To vyústilo v lehkou změnu toho, jak jsi vyřešil tu dynamičnost programových priorit a nějaký drobný sémantický a ARIA změny.

Změnil jsem taky to logo v Open Graph. Ty jsi tam měl čtvercové. Různě sítě používají různé poměry a jsou různě flexibilní. Já jsem vzal aktuální poměr, který používá Facebook v doporučovaných rozměrech 1200 × 630.

Může se zdát, že jsem toho změnil hodně, ale bez toho tvýho základu by to, Josefe, nešlo. Protože ty změny do všeho ale hrabaly celkem na bazální úrovni, tak je ten tvůj původní commit přepsal.

f4z4on commented:
Na psaní těch zpráv ke Git commitům je dobrý tenhle [článek od Chrise Beamse](https://chris.beams.io/posts/git-commit/). Je to delší, ale odkazuje tam na dobrý příklady a bere to víc komplexně než jak tom píše většina lidí (resp. spoustu užitečnejch tipů, který lidi tak nějakou trousej po webu, dává do toho jednoho svého článku).

jusephe committed:
Update og:image:width and height in kontakt.html

jusephe commented:
Díky za všechna ta vylepšení! Uvědomuju si, že spousta věcí byla trochu odfláknutých, jen aby to nějak bylo (a i tak mi to zabralo docela dost času), ale dělal jsem to s vědomím toho, že za pár týdnů to celé nahradíme úplně novým webem, tak jsem to moc nehrotil... ani commit message :D
Na ten článek se podívám, dík

f4z4on commented:
Však jsme se o tom i tak bavili a je to úplně v pořádku. Já už ani nevím, co ve mě spustilo tu reakci se na to podívat a nějak jsem se do toho ponořil… 😅

Dík za opravu…

f4z4on committed:
Remove executable flags

All the files are static. There’s no need for such permissions.

f4z4on committed:
Change page title position a little

Because the first letter of the title is capital P and because of some
of weirdness in Lato, the heading is visibly not aligned with the rest
of the text on the page. Moving it somewhere between 2 and 3 pixels to
the left almost does the trick which is enough for now.

f4z4on committed:
Removed redundant files

Files are deployed to S3-like object storage with load balancer in front of it (to serve from apex domain). There’s no Apache HTTP Server nor anything that can understand .htaccess.

Gitignore file was a leftover from previous Create React App project.
